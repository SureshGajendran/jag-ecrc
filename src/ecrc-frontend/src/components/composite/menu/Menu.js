import React from "react";
import PropTypes from "prop-types";

import "./Menu.css";
import MenuItem from "../../base/menuItem/MenuItem";

export default function Menu({ menuItems }) {
  // If menuItems are provided, create an array of MenuItems, else default to empty array
  const menuList = menuItems
    ? menuItems.map((menuItem, index) => {
        return (
          <MenuItem
            key={menuItem.name}
            url={menuItem.url}
            name={menuItem.name}
          />
        );
      })
    : [];

  return (
    <div className="col-sm-3 col-md-3">
      <nav className="navbar navbar-expand-md navbar-light" role="navigation">
        <button
          className="navbar-toggler"
          type="button"
          data-toggle="collapse"
          data-target="#leftNav"
          aria-controls="leftNav"
          aria-expanded="false"
          aria-label="Toggle left navigation"
        >
          <span className="navbar-toggler-icon" />
        </button>

        <div id="leftNav" className="collapse navbar-collapse">
          <ul className="menuList navbar-nav flex-column">{menuList}</ul>
        </div>
      </nav>
    </div>
  );
}

Menu.propTypes = {
  menuItems: PropTypes.array
};
