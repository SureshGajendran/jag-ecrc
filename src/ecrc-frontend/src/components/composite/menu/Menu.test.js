import React from "react";
import { create } from "react-test-renderer";

import Menu from "./Menu";

describe("Menu Component", () => {
  test("Matches the snapshot", () => {
    const menuItems = [
      {
        name: "Home",
        url: "/"
      },
      {
        name: "Somewhere",
        url: "/somewhere"
      },
      {
        name: "Somewhere else",
        url: "/somewhereelse"
      },
      {
        name: "Somewhere with a long name for no reason",
        url: "/here"
      }
    ];

    const menu = create(<Menu menuItems={menuItems} />);
    expect(menu.toJSON()).toMatchSnapshot();
  });
});
