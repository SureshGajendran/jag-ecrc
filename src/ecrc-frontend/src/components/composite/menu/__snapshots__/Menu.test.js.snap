// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Component Matches the snapshot 1`] = `
<div
  className="col-sm-3 col-md-3"
>
  <nav
    className="navbar navbar-expand-md navbar-light"
    role="navigation"
  >
    <button
      aria-controls="leftNav"
      aria-expanded="false"
      aria-label="Toggle left navigation"
      className="navbar-toggler"
      data-target="#leftNav"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="leftNav"
    >
      <ul
        className="navbar-nav flex-column"
      >
        <li
          className="menuItem"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          className="menuItem"
        >
          <a
            href="/somewhere"
          >
            Somewhere
          </a>
        </li>
        <li
          className="menuItem"
        >
          <a
            href="/somewhereelse"
          >
            Somewhere else
          </a>
        </li>
        <li
          className="menuItem"
        >
          <a
            href="/here"
          >
            Somewhere with a long name for no reason
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
